tests/it/scargo_test/it_scargo_commands_flow.py

```diff
@@ -0,0 +1,256 @@
+import os
+from pathlib import Path
+from shutil import copy, copytree
+
+import pytest
+from utils import (
+    ScargoRunner,
+    add_profile_to_toml,
+    assert_str_in_CMakeLists,
+    assert_str_in_file,
+    get_bin_name,
+    get_copyright_text,
+    get_project_name,
+)
+
+from scargo import cli
+
+FIX_TEST_FILES_PATH = Path(
+    pytest.it_path, "test_projects", "test_files", "fix_test_files"
+)
+
+IDF_SDKCONFIG_FILE_PATH = Path(
+    pytest.it_path, "test_projects", "test_files", "esp_32_idf_config", "sdkconfig"
+)
+
+PROJECT_CREATION_x86 = [
+    "new_project_x86",
+    "copy_project_x86",
+]
+
+PROJECT_CREATION_esp32 = [
+    "new_project_esp32",
+    "copy_project_esp32",
+]
+
+PROJECT_CREATION_stm32 = [
+    "new_project_stm32",
+    "copy_project_stm32",
+]
+
+
+@pytest.fixture()
+def new_project_x86():
+    # Arrange
+    runner = ScargoRunner()
+
+    # New Help
+    result = runner.invoke(cli, ["new", "-h"])
+    assert result.exit_code == 0
+
+    # New
+    result = runner.invoke(cli, ["new", pytest.new_test_project_name, "--target=x86"])
+    bin_name = get_bin_name()
+    expected_bin_file_path = Path("./src/", bin_name.lower() + ".cpp")
+    assert expected_bin_file_path.exists()
+    assert result.exit_code == 0
```

Comment:
I'd rather see reversed asserts and use is_file
```suggestion
    assert result.exit_code == 0
    assert expected_bin_file_path.is_file()
```